FinancialTracker/ViewControllers/BalanceViewController.swift

```diff
@@ -85,7 +86,7 @@ class BalanceViewController: UIViewController {
     }
     
     @objc func signOut() {
-        FirebaseHandler.shared.signOut { firebaseError, _ in
+        databaseManager?.authManager?.signOut { firebaseError, _ in
```

Comment:
Again, either use `authManager` which hosts a private `databaseManager`, or use them separately in an explicit way. 
But it is not looking good now because in one place you have `authManager.databaseManager` and then in another place you have `databaseManager.authManager`. It is actually not only not looking good but you are setting yourself up for a **classic circular reference bug**.